integreat_compass/cms/views/interactions/vote_actions.py

```diff
@@ -0,0 +1,35 @@
+from django.shortcuts import redirect
+from django.views.decorators.http import require_POST
+
+from ...decorators import permission_required
+from ...models import Vote
+
+
+@require_POST
+@permission_required("vote_on_offer")
+def add_vote(request, offer_version_id):
+    """
+    Add vote of a user to an offer version
+
+    :param request: Object representing the user call
+    :type request: ~django.http.HttpRequest
+
+    :param offer_version_id: internal id of the offer_version to be voted on
+    :type offer_version_id: int
+
+    :return: The rendered template response
+    :rtype: ~django.template.response.TemplateResponse
+    """
+    approval = request.POST.get("approval")
+    vote = Vote.objects.filter(creator=request.user).filter(
+        offer_version=offer_version_id
+    )
+    if vote:
+        vote[0].approval = approval
+        vote[0].save()
+    else:
+        new_vote = Vote(
+            creator=request.user, offer_version_id=offer_version_id, approval=approval
+        )
+        new_vote.save()
```

Comment:
```suggestion
    vote = Vote.objects.filter(
        creator=request.user, offer_version=offer_version_id
    ).first()
    if vote:
        vote.approval = approval
        vote.save()
    else:
        Vote.objects.create(
            creator=request.user, offer_version_id=offer_version_id, approval=approval
        )

```

Reply:
```suggestion
    vote, created = Vote.objects.update_or_create(
        creator=request.user,
        offer_version_id=offer_version_id,
        defaults={"approval": approval},
    )
```
Why not simply like this?

Reply:
Nice, did not know about the update_or_create function :+1: